internal/plumbing/ticks.go

```diff
@@ -0,0 +1,158 @@
+	return []core.ConfigurationOption{{
+		Name:        ConfigTicksSinceStartTickSize,
+		Description: "How long each 'tick' represents in hours.",
+		Flag:        "tick-size",
+		Type:        core.IntConfigurationOption,
+		Default:     DefaultTicksSinceStartTickSize},
+	}
+}
+
+// Configure sets the properties previously published by ListConfigurationOptions().
+func (ticks *TicksSinceStart) Configure(facts map[string]interface{}) error {
+	if val, exists := facts[ConfigTicksSinceStartTickSize].(int); exists {
+		ticks.tickSize = time.Duration(val) * time.Hour
+	} else {
+		// default to 1 day
+		ticks.tickSize = 24 * time.Hour
+	}
+	if ticks.commits == nil {
+		ticks.commits = map[int][]plumbing.Hash{}
+	}
+	facts[FactCommitsByTick] = ticks.commits
+	return nil
+}
+
+// Initialize resets the temporary caches and prepares this PipelineItem for a series of Consume()
+// calls. The repository which is going to be analysed is supplied as an argument.
+func (ticks *TicksSinceStart) Initialize(repository *git.Repository) error {
+	ticks.tick0 = &time.Time{}
+	ticks.previousTick = 0
+	if len(ticks.commits) > 0 {
+		keys := make([]int, len(ticks.commits))
+		for key := range ticks.commits {
+			keys = append(keys, key)
+		}
+		for _, key := range keys {
+			delete(ticks.commits, key)
+		}
+	}
+	if r, err := repository.Remotes(); err == nil && len(r) > 0 {
+		ticks.remote = r[0].Config().URLs[0]
+	}
+	return nil
+}
+
+// Consume runs this PipelineItem on the next commit data.
+// `deps` contain all the results from upstream PipelineItem-s as requested by Requires().
+// Additionally, DependencyCommit is always present there and represents the analysed *object.Commit.
+// This function returns the mapping with analysis results. The keys must be the same as
+// in Provides(). If there was an error, nil is returned.
+func (ticks *TicksSinceStart) Consume(deps map[string]interface{}) (map[string]interface{}, error) {
+	commit := deps[core.DependencyCommit].(*object.Commit)
+	index := deps[core.DependencyIndex].(int)
+	if index == 0 {
+		// first iteration - initialize the file objects from the tree
+		// our precision is 1 day
+		*ticks.tick0 = commit.Committer.When
+		if ticks.tick0.Unix() < 631152000 { // 01.01.1990, that was 30 years ago
+			log.Println()
+			log.Printf("Warning: suspicious committer timestamp in %s > %s: %d",
+				ticks.remote, commit.Hash.String(), ticks.tick0.Unix())
+		}
+	}
+
+	tick := int(commit.Committer.When.Sub(*ticks.tick0) / ticks.tickSize)
+	if tick < ticks.previousTick {
+		// rebase works miracles, but we need the monotonous time
+		tick = ticks.previousTick
+	}
+
+	ticks.previousTick = tick
+	tickCommits := ticks.commits[tick]
+	if tickCommits == nil {
+		tickCommits = []plumbing.Hash{}
+	}
+
+	exists := false
+	if commit.NumParents() > 0 {
+		for i := range tickCommits {
+			if tickCommits[len(tickCommits)-i-1] == commit.Hash {
+				exists = true
```

Comment:
The original code was written a long time ago when I was still learning Go. We should definitely `break` here after setting `exists` to ` true`.

Reply:
👌 b4af94f